invenio_checks/components.py

```diff
@@ -95,11 +95,17 @@ def _run_checks(self, identity, data=None, record=None, errors=None, **kwargs):
                 ]
                 errors.extend(check_errors)
 
+                # Delete previous draft check run if this is a publish action
+                if not record.is_draft:
+                    CheckRun.query.filter_by(
+                        config_id=check_config.id, record_id=record.id, is_draft=True
+                    ).delete()
+
                 latest_check = (
                     CheckRun.query.filter(
                         CheckRun.config_id == check_config.id,
                         CheckRun.record_id == record.id,
-                        CheckRun.is_draft.is_(True),
+                        CheckRun.is_draft.is_(record.is_draft),
```

Comment:
See above, but this becomes especially confusing now. Is `record` a record or draft?  We shouldn't rely on calling `record.is_draft`, it should be obvious/readable from the context we're in.